fdbserver/storageserver.actor.cpp

```diff
@@ -4625,7 +4637,10 @@ ACTOR Future<Void> fetchKeys(StorageServer* data, AddingShard* shard) {
 		TraceEvent(SevDebug, "FKUpdateBatch", data->thisServerID).detail("FKID", interval.pairID);
 
 		shard->phase = AddingShard::Waiting;
+		while (!shard->updates.empty() && shard->updates[0].version <= fetchVersion)
```

Comment:
move the code block up to where the fetch version is set, we know once we set the fetch version we will never add any more updates at a lower version

Reply:
Just wondering whether there has inconsistency problems, like `AddingShard::addMutation` doesn't read the updated value of shard->fetchVersion because of caching. Actually we already clean up the updates where the fetch version is set. The intention of this line here to do the double check.